processing/src/main/java/org/apache/druid/timeline/CompactionState.java

```diff
@@ -146,4 +149,29 @@ public String toString()
            ", metricsSpec=" + metricsSpec +
            '}';
   }
+
+  public static Function<Set<DataSegment>, Set<DataSegment>> compactionStateAnnotateFunction(
```

Comment:
Should we just call this `addCompactionStateToSegments()`
"Annotate" seems vague.